packages/server/graphql/public/mutations/getTemplateSuggestion.ts

```diff
@@ -0,0 +1,65 @@
+import {getUserId} from '../../../utils/authorization'
+import OpenAIServerManager from '../../../utils/OpenAIServerManager'
+import standardError from '../../../utils/standardError'
+import isValid from '../../isValid'
+import {MutationResolvers} from '../resolverTypes'
+
+type Prompt = {
+  question: string
+  description: string
+}
+
+type PromptsByTemplate = {
+  [templateId: string]: Prompt[]
+}
+
+const getTemplateSuggestion: MutationResolvers['getTemplateSuggestion'] = async (
+  _source,
+  {prompt},
+  {authToken, dataLoader}
+) => {
+  const viewerId = getUserId(authToken)
+  const organizationUsers = await dataLoader.get('organizationUsersByUserId').load(viewerId)
+  const userOrgIds = organizationUsers.map(({orgId}) => orgId)
+  const availableOrgIds = ['aGhostOrg', ...userOrgIds]
+  const allTemplates = (await dataLoader.get('meetingTemplatesByOrgId').loadMany(availableOrgIds))
+    .filter(isValid)
+    .flat()
+
+  const templateIds = allTemplates.map((template) => template.id)
+  const prompts = (await dataLoader.get('reflectPromptsByTemplateId').loadMany(templateIds))
+    .filter(isValid)
+    .flat()
+
+  const promptsByTemplate: PromptsByTemplate = {}
+  prompts.forEach((prompt) => {
+    const {templateId} = prompt
+    if (!promptsByTemplate[templateId]) {
+      promptsByTemplate[templateId] = []
+    }
+    promptsByTemplate[templateId]!.push({
+      question: prompt.question,
+      description: prompt.description
+    })
+  })
+
+  const templatesWithPrompts = allTemplates.map((template) => ({
+    templateId: template.id,
+    templateName: template.name,
+    prompts: promptsByTemplate[template.id] || []
+  }))
+
+  const manager = new OpenAIServerManager()
+  const templateRes = await manager.getTemplateSuggestion(templatesWithPrompts, prompt)
```

Comment:
I got a server timeout error because an OpenAI query took a long time, and the mutation didn't respond in time.

A few options:
- Set a timeout limit on the OpenAI query, but that's not ideal as the query will fail when it could successfully return the response with a little more time
- Speed up the OpenAI query
- Increase the timeout window for the server mutation

Is there a way to increase the timeout window for just one server mutation?

Reply:
Sounds good. I've created an issue to implement this so we can start playing with the feature sooner: https://github.com/ParabolInc/parabol/issues/9340